beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -265,6 +406,70 @@ impl<E: EthSpec> RangeBlockComponentsRequest<E> {
                     }
                 }
 
+                Some(resp)
+            }
+            // Reuse same logic that we use for coupling data columns for now.
+            // todo(pawan): we should never get a coupling error here, so simplify this
```

Comment:
What do you mean here? 
This function is getting a bit long, we can probably extract the logic to a separate function for readability?